src/protocols/rdp/channels/rdpsnd/rdpsnd-messages.c

```diff
@@ -266,6 +307,14 @@ void guac_rdpsnd_wave_handler(guac_rdp_common_svc* svc,
 
     guac_rdp_client* rdp_client = (guac_rdp_client*) client->data;
     guac_audio_stream* audio = rdp_client->audio;
+    
+    /* Verify we have at least 4 bytes, which is manually copied below. */
+    if (Stream_Length(input_stream) < 4) {
```

Comment:
That 4 bytes is actually copied from `rdpsnd->initial_wave_data`, not the `input_stream`. For `input_stream`, the read happens when `guac_audio_stream_write_pcm()` is called, and reads `rdpsnd->incoming_wave_size + 4` bytes:

https://github.com/apache/guacamole-server/blob/9c37fc56171d55b8fa838949d7fd40db85506857/src/protocols/rdp/channels/rdpsnd/rdpsnd-messages.c#L281-L282

Reply:
So, remove this entirely, or we should be checking `rdpsnd->initial_wave_data`?

Reply:
We should be ensuring that `Stream_Length(input_stream)` is at least `rdpsnd->incoming_wave_size + 4`.

Reply:
Got it.  Done.